sonar-db/src/main/java/org/sonar/db/component/ComponentDto.java

```diff
@@ -19,33 +19,88 @@
  */
 package org.sonar.db.component;
 
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
 import java.util.Date;
+import java.util.List;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 import org.apache.commons.lang.builder.ToStringBuilder;
 import org.sonar.api.component.Component;
 import org.sonar.api.resources.Scopes;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static org.sonar.db.component.ComponentValidator.checkComponentKey;
 import static org.sonar.db.component.ComponentValidator.checkComponentName;
 
 public class ComponentDto implements Component {
 
-  public static final String MODULE_UUID_PATH_SEP = ".";
+  public static final String UUID_PATH_SEPARATOR = ".";
```

Comment:
I don't think so. DTOs are the classes the closest from relational model. For instance that's why I added documentation about columns on the DTO fields. Having the constants here makes sense IMO.
About methods (in fact only `getUuidPathAsList()`), I don't care as it's used only in a single place. Changing visibility to package protected sounds enough. No need for moving it to a static method of DAO.